src/tsdb.c

```diff
@@ -224,75 +224,85 @@ int SeriesAddSample(Series *series, api_timestamp_t timestamp, double value) {
     return TSDB_OK;
 }
 
-SeriesIterator SeriesQuery(Series *series, api_timestamp_t minTimestamp, api_timestamp_t maxTimestamp) {
-    SeriesIterator iter;
+int SeriesQuery(Series *series, SeriesIterator *iter, 
+                            api_timestamp_t minTimestamp, api_timestamp_t maxTimestamp) {
     timestamp_t rax_key;
-    iter.series = series;
+    iter->series = series;
+    ChunkFuncs *funcs = iter->series->funcs;
+    
     // get the rightmost chunk whose base timestamp is smaller or equal to minTimestamp
     seriesEncodeTimestamp(&rax_key, minTimestamp);
-    iter.dictIter = RedisModule_DictIteratorStartC(series->chunks, "<=", &rax_key, sizeof(rax_key));
-
-    // if no such chunk exists, we will start the search from the first chunk
-    if (!RedisModule_DictNextC(iter.dictIter, NULL, (void*)&iter.currentChunk))
-    {
-        RedisModule_DictIteratorReseekC(iter.dictIter, "^", NULL, 0);
-        RedisModule_DictNextC(iter.dictIter, NULL, (void*)&iter.currentChunk);
+    iter->dictIter = RedisModule_DictIteratorStartC(series->chunks, "<=", &rax_key, sizeof(rax_key));
+    RedisModule_DictNextC(iter->dictIter, NULL, (void*)&iter->currentChunk);
```

Comment:
@ashtul you need to check the return value of  `RedisModule_DictNextC` before using it.